nomad/structs/bitmap.go

```diff
@@ -76,3 +80,25 @@ func (b Bitmap) IndexesInRange(set bool, from, to uint) []int {
 
 	return indexes
 }
+
+// IndexesInRangeFiltered returns the indexes in which the values are either set
+// or unset based on the passed parameter in the passed range, and do not appear
+// in the filter slice
+func (b Bitmap) IndexesInRangeFiltered(set bool, from, to uint, filter []int) []int {
+	var indexes []int
+	for i := from; i <= to && i < b.Size(); i++ {
+		c := b.Check(i)
+		if c && set || !c && !set {
```

Comment:
```suggestion
		if c == set {
```
but it might be best to just match the above func and leave it be.

Reply:
Wow this got copy-and-pasted from the other function and I didn't even notice how weird a thing to do this is. Will fix.